### ISSUE TITLE
Data Import Error on Data Type Change for Non-URL Values in Columns 

### ISSUE THREAD

**Issue by ghislaineguerin** [mathesar-foundation/mathesar#3512]
## Description
In data import, if a column contains non-url values, such as the following:
<img width="169" alt="image" src="https://github.com/mathesar-foundation/mathesar/assets/845767/be4bf345-0f1e-4d5b-9b6d-aec40c1b2f5e">

Attempting to update the data type under these circumstances results in a crash, accompanied by the following error message:
<img width="634" alt="image" src="https://github.com/mathesar-foundation/mathesar/assets/845767/14e343b4-2ea2-4588-aebc-42f27b7448fd">

## Expected behavior
When attempting a data type change for such a column, the operation should either complete successfully, or a clear notification should be provided, indicating that the data type change cannot be executed.

**Comment by hitenvidhani** [comment]
Found a similar error 
[Video](https://github.com/mathesar-foundation/mathesar/assets/60215053/c70bef3d-1c82-46b1-b87c-6437806a2013), is this behaviour expected or we want to show a clear warning message here for the user to understand?

**Comment by kgodey** [comment]
Tagging @seancolsen to take a look at @hitenvidhani's comment.

**Comment by seancolsen** [comment]
@hitenvidhani I agree it would be nice to improve the error message show in your attached video. Please open a separate ticket to track that though. It's a different issue than the one described here, and I don't want to add confusion to this thread. In general, please err on the side of opening new tickets for things like this, or ask in Matrix in cases where you're not sure if something warrants a ticket.

**Comment by mathemancer** [comment]
@ghislaineguerin Have you looked at this recently? It now does the following:

Upload:

![image](https://github.com/user-attachments/assets/773d9b48-6d6a-4708-8f62-370abceaf46c)

Change to URI type:

![image](https://github.com/user-attachments/assets/f5e4fbcf-8c5e-4623-90ad-00213ba3aa17)

See an error, with some options for proceeding:

![image](https://github.com/user-attachments/assets/c36afc3c-78fc-470b-9d2e-89cd37363513)

I do think the error message could be tidied up, but the relevant bit ("N/A isn't a URI") is prominent. Also, the buttons at the bottom work properly and let the user proceed.

I suggest we close this issue, and make a general project (probably for after beta launch) to reduce the verbosity of error messages when possible.

**Comment by ghislaineguerin** [comment]
@mathemancer that makes sense. I agree with closing the issue and making a general project.
